### ISSUE TITLE
Define the new package high-level API (structure)

### ISSUE THREAD

**Issue by oesteban** [nipreps/nireports#25]
Before we dive into porting (#3, #22) we need to agree on a package structure. What about something like this?

```
nireports
 |-- nireports.assembler (aggregates reportlets into an HTML document)
 |    |-- nireports.assembler.data (contains the default template)
 |    |-- nireports.assembler.pyodide (an assembler that runs on the browser / perhaps a separate package?)
 |    |-- nireports.assembler.report (contains the `Report` class and related)
 |    `-- nireports.assembler.tools (contains `generate_reports` and `run_reports`)
 |-- nireports.reportlets
 |    |-- nireports.reportlets.animated (registration plots, animated with JS - use the other components)
 |    |-- nireports.reportlets.components -> niworkflows.viz.utils.py (foundational tooling)
 |    |-- nireports.reportlets.composite (static reportlets combining static reportlets, e.g., the fMRIPlot)
 |    |-- nireports.reportlets.mosaic (segmentation plots with contours, but not moving)
 |    |-- nireports.reportlets.misc (e.g., the carpetplot)
 |    `-- nireports.reportlets.notebook (the notebook wrapper -convert to decorator?- for animated reportlets)
 `-- nireports.export (some tooling to zip reports, and maybe share them somehow e.g., email)
```

Notes:

* `nireports.assembler.tools` needs to rename internal functions to indicate "types" of reports (I can think of "participant" vs. "group").

**Comment by oesteban** [comment]
perhaps `nireports.export` -> `nireports.assembler.export`?

**Comment by oesteban** [comment]
One other question is whether the process of porting should be done with a "neutral" user so we do not pollute the credits generation system (or complicate it by marking commits as non-contributing).

**Comment by oesteban** [comment]
WDY'all think?

cc/ @ZviBaratz @effigies

**Comment by ZviBaratz** [comment]
> What about something like this?

LGTM. I don't really have a deep enough understanding of the existing code-base and future plans to think about critically, but it looks reasonable. 

> perhaps nireports.export -> nireports.assembler.export?

I actually liked the first better.

> One other question is whether the process of porting should be done with a "neutral" user (e.g., nipreps-bot) so we do not pollute the credits generation system (or complicate it by marking commits as non-contributing).

That sounds like a good idea, I also wasn't sure how to approach this. Tried looking up how to check out only particular files from one git archive to another but couldn't find anything. 

IMO it will be easiest to port everything we can as it is and then refactor incrementally while trying not to break anything (or a minimal fraction of things, at least).

**Comment by oesteban** [comment]
Note: added `nireports.reportlets.interfaces`.

**Comment by eilidhmacnicol** [comment]
Based on @celprov 's notes, this is the code that should be imported:

`niworkflows.viz.notebooks` -> `assembler`
`niworkflows.viz.plots` -> `reportlets`

- [ ] `fMRIplot` -> `reportlets.composite`
- [ ] `plot_carpet` -> `reportlets.misc`
- [ ] `spikesplot` -> `reportlets.composite`
- [ ] `spikes_plot_cb` -> `reportlets.composite`
- [ ] `confound_plot` -> `reportlets.components`
- [ ] `compcorr.variance.plot` -> `reportlets.components`

@ZviBaratz this hopefully provides some framework for where these tools can go, but I think if necessary we can also have a discussion about it if someone has particularly strong feelings about where a plot should go.

**Comment by celprov** [comment]
That is the code that should be imported from mriqc :
`mriqc.interfaces.viz` -> `reportlets.interfaces`
`mriqc.viz.svg` -> `assembler`
`mriqc.viz.utils` -> `reportlets`

- [ ] `plot_slice` -> `reportlets.components`
- [ ] `plot_slice_tern` ->  `reportlets.components`
- [ ] `plot_spikes` -> ??
- [ ] `plot_mosaic` -> `reportlets.mosaic`
- [ ] `plot_fd` -> It seems it is not used so we should not copy it
- [ ] `plot_dist` -> not used so no copy
- [ ] `plot_vline` -> called by plot_fd & plot_dist, but as the latter are not used, it should not be copied a priori
- [ ] `plot_segmentation` -> `reportlets.mosaic`
- [ ] `plot_qi2` -> `reportlets.misc`

**Comment by oesteban** [comment]
> That is the code that should be imported from mriqc : `mriqc.interfaces.viz` -> `reportlets.interfaces` `mriqc.viz.svg` -> `assembler` `mriqc.viz.utils` -> `reportlets`
> 
>     * [x]  `plot_slice` -> `reportlets.components`
> 
>     * [x]  `plot_slice_tern` ->  `reportlets.components`
> 
>     * [ ]  `plot_spikes` -> ??
> 
>     * [x]  `plot_mosaic` -> `reportlets.mosaic`
> 
>     * [ ]  `plot_fd` -> It seems it is not used so we should not copy it
> 
>     * [ ]  `plot_dist` -> not used so no copy
> 
>     * [ ]  `plot_vline` -> called by plot_fd & plot_dist, but as the latter are not used, it should not be copied a priori
> 
>     * [ ]  `plot_segmentation` -> `reportlets.mosaic`
> 
>     * [ ]  `plot_qi2` -> `reportlets.misc`

Yeah, there's a bunch of things we don't need to port (starting with those not used). I'm thinking that the segmentation plotter should just be replaced by the default one.
